core/src/main/java/org/infinispan/cache/impl/CacheImpl.java

```diff
@@ -129,6 +129,8 @@
 @MBean(objectName = CacheImpl.OBJECT_NAME, description = "Component that represents an individual cache instance.")
 public class CacheImpl<K, V> implements AdvancedCache<K, V> {
    public static final String OBJECT_NAME = "Cache";
+   private static final EnumSet<Flag> IGNORE_RETURN_VALUES_FLAG_SET = EnumSet.of(IGNORE_RETURN_VALUES);
```

Comment:
If I made it immutable then it wouldn't be an `EnumSet` any more, and the bulk `EnumSet` methods work better when both sets are `EnumSets`.

I'm not sure it's worth it TBH, at first I thought I could remove all the flag additions in the interceptors but it didn't quite work out. I expect the number of allocations would be about the same either way, and since I already had this implemented I didn't try making a copy at the beginning.